internal/attestation/aws/validator.go

```diff
@@ -21,23 +20,31 @@ import (
 	"github.com/google/go-tpm/tpm2"
 )
 
-// Issuer for AWS TPM attestation
+// Validator for AWS TPM attestation.
 type Validator struct {
 	oid.AWS
 	*vtpm.Validator
+	metadataClient awsMetadataAPI
 }
 
+// NewValidator create a new Validator structure and returns it.
 func NewValidator(pcrs map[uint32][]byte, enforcedPCRs []uint32, log vtpm.WarnLogger) *Validator {
-	return &Validator{
-		Validator: vtpm.NewValidator(
-			pcrs,
-			enforcedPCRs,
-			getTrustedKey,
-			validateCVM,
-			vtpm.VerifyPKCS1v15,
-			log,
-		),
+	v := &Validator{}
+	v.Validator = vtpm.NewValidator(
+		pcrs,
+		enforcedPCRs,
+		getTrustedKey,
+		v.tpmEnabled,
+		vtpm.VerifyPKCS1v15,
+		log,
+	)
+	client, err := config.LoadDefaultConfig(context.Background(), config.WithEC2IMDSRegion())
+	if err != nil {
+		return nil
 	}
```

Comment:
I didn't consider the implications for existing code if we create the client in the Validator init function...
In a lot of places our code assumes the `NewValidator` functions cannot return errors and the returned value is always valid (e.g. not nil).

Made a suggestion PR to fix this, please check it